frontend/pages/settings/dlna.vue

```diff
@@ -0,0 +1,249 @@
+    const dlnaSettings = (
+      await $api.configuration.getNamedConfiguration({ key: 'dlna' })
+    ).data;
+
+    const dlnaProfiles = (await $api.dlna.getProfileInfos()).data;
+    const users = (await $api.user.getUsers()).data;
+
+    return { dlnaSettings, dlnaProfiles, users };
+  },
+  data() {
+    return {
+      dlnaSettings: {} as DlnaNamedConfiguration,
+      dlnaProfiles: [] as DeviceProfileInfo[],
+      users: [] as UserDto[],
+      selectedProfile: {} as DeviceProfile,
+      deviceInfoDialog: false,
+      isCustomProfile: false,
+      currentProfileId: ''
+    };
+  },
+  computed: {
+    headers(): { text: string; value: string }[] {
+      return [
+        {
+          text: this.$t('settings.dlna.profile.name'),
+          value: 'Name'
+        }
+      ];
+    },
+    userProfiles(): DeviceProfileInfo[] {
+      return this.dlnaProfiles.filter(
+        (profile) => profile.Type === DeviceProfileType.User
+      );
+    },
+    systemProfiles(): DeviceProfileInfo[] {
+      return this.dlnaProfiles.filter(
+        (profile) => profile.Type === DeviceProfileType.System
+      );
+    },
+    bindToRelevantConfig(): RelevantConfig {
+      return {
+        EnablePlayTo: this.dlnaSettings.EnablePlayTo,
+        EnableDebugLog: this.dlnaSettings.EnableDebugLog,
+        ClientDiscoveryIntervalSeconds:
+          this.dlnaSettings.ClientDiscoveryIntervalSeconds,
+        EnableServer: this.dlnaSettings.EnableServer,
+        BlastAliveMessages: this.dlnaSettings.BlastAliveMessages,
+        AliveMessageIntervalSeconds:
+          this.dlnaSettings.AliveMessageIntervalSeconds,
+        DefaultUserId: this.dlnaSettings.DefaultUserId
+      };
+    }
+  },
+  methods: {
+    async storeDlnaConfiguration(): Promise<void> {
+      await this.$api.configuration.updateNamedConfiguration(
+        {
+          key: 'dlna'
+        },
+        { data: this.dlnaSettings }
+      );
+    },
+    async setSelectedDevice(selectedDevice: DeviceProfileInfo): Promise<void> {
+      if (!isNil(selectedDevice)) {
+        this.currentProfileId = selectedDevice.Id as string;
+        this.selectedProfile = (
+          await this.$api.dlna.getProfile({
+            profileId: selectedDevice.Id as string
+          })
+        ).data;
+        this.isCustomProfile = this.userProfiles.includes(selectedDevice);
+        this.deviceInfoDialog = true;
+      }
+    },
+    async saveProfile(profile: DeviceProfile): Promise<void> {
+      await this.$api.dlna.updateProfile({
+        profileId: this.currentProfileId,
+        deviceProfile: profile
+      });
+      this.dlnaProfiles = (await this.$api.dlna.getProfileInfos()).data;
+      this.closeDialog();
+    },
+    async deleteSelectedProfile(): Promise<void> {
+      await this.$api.dlna.deleteProfile({
+        profileId: this.currentProfileId
+      });
+      this.dlnaProfiles = (await this.$api.dlna.getProfileInfos()).data;
+      this.closeDialog();
+    },
```

Comment:
Are we giving any visual hint that these functions are running? I.E an spinning circle, an snackbar message, etc...

Reply:
We don't currently, however, I can include a snackbar message